pkg/workloadmeta/collectors/internal/containerd/image.go

```diff
@@ -116,22 +190,29 @@ func (c *collector) handleImageEvent(ctx context.Context, containerdEvent *conta
 		return c.handleImageCreateOrUpdate(ctx, containerdEvent.Namespace, event.Name, nil)
 
 	case imageDeletionTopic:
+		c.handleImagesMut.Lock()
+
 		event := &events.ImageDelete{}
 		if err := proto.Unmarshal(containerdEvent.Event.Value, event); err != nil {
+			c.handleImagesMut.Unlock()
 			return fmt.Errorf("error unmarshaling containerd event: %w", err)
 		}
 
 		imageID, found := c.knownImages.getImageID(event.Name)
 		if !found {
+			c.handleImagesMut.Unlock()
 			return nil
 		}
```

Comment:
Can we use `defer` instead of unlocking the mutex before each `return` statement ?

```suggestion
	case imageDeletionTopic:
		c.handleImagesMut.Lock()
		defer c.handleImagesMut.Unlock()

		event := &events.ImageDelete{}
		if err := proto.Unmarshal(containerdEvent.Event.Value, event); err != nil {
			return fmt.Errorf("error unmarshaling containerd event: %w", err)
		}

		imageID, found := c.knownImages.getImageID(event.Name)
		if !found {
			return nil
		}
```

Reply:
Not in this case because this function calls `c.handleImageCreateOrUpdate` in some cases, and that function also takes the lock.